app/services/resource_filter.rb

```diff
@@ -70,6 +70,12 @@ def filter(collection)
         proposal_in_meetings_ids = MeetingProposal.pluck(:proposal_id).uniq
         collection = collection.where(id: proposal_in_meetings_ids)
       end
+
+      if @params["source"] && @params["source"].include?("organization")
+        proposal_from_organizations_ids = Organization.verified.map{ |o| o.user.proposals.ids }.flatten.uniq
```

Comment:
This might work, as it would use two subqueries:

```ruby
Proposal.where(author: User.joins(:organization).where(organization: Organization.verified))
```

There's much space for improvement but I don't think it'd be a problem with such a small dataset.